cli/start.go

```diff
@@ -90,75 +64,79 @@ func (r *RootCmd) start() *clibase.Cmd {
 			return nil
 		},
 	}
-	return cmd
-}
 
-type prepStartWorkspaceArgs struct {
-	Action            WorkspaceCLIAction
-	TemplateVersionID uuid.UUID
+	cmd.Options = append(cmd.Options, parameterFlags.allOptions()...)
 
-	LastBuildParameters []codersdk.WorkspaceBuildParameter
+	return cmd
+}
 
-	PromptBuildOptions bool
-	BuildOptions       []codersdk.WorkspaceBuildParameter
+type startWorkspaceArgs struct {
+	workspace      codersdk.Workspace
+	parameterFlags workspaceParameterFlags
+	action         WorkspaceCLIAction
 }
 
-func prepStartWorkspace(inv *clibase.Invocation, client *codersdk.Client, args prepStartWorkspaceArgs) ([]codersdk.WorkspaceBuildParameter, error) {
-	ctx := inv.Context()
+func buildWorkspaceStartRequest(inv *clibase.Invocation, client *codersdk.Client, args startWorkspaceArgs) (codersdk.CreateWorkspaceBuildRequest, error) {
+	version := args.workspace.LatestBuild.TemplateVersionID
+	if args.workspace.AutomaticUpdates == codersdk.AutomaticUpdatesAlways || args.action == WorkspaceUpdate {
+		template, err := client.Template(inv.Context(), args.workspace.TemplateID)
+		if err != nil {
+			return codersdk.CreateWorkspaceBuildRequest{}, xerrors.Errorf("get template: %w", err)
+		}
+		version = template.ActiveVersionID
```

Comment:
workspace already has a TemplateActiveVersionID --- so I think you can skip the API call to get the template